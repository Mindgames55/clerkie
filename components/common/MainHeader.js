'use client'

import { useSelectedLayoutSegment } from 'next/navigation';
import styles from './MainHeader.module.css'

export default function MainHeader() {
    const segment = useSelectedLayoutSegment();

    const pageHeader = !segment ? 'Home' : 'Friends';
    return (
        <div className={styles.container}>
            <h1 className={styles.heading}>{pageHeader}</h1>
        </div>
    )
}