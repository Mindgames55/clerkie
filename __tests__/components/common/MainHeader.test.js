import { useSelectedLayoutSegment } from 'next/navigation';
import { render, screen } from '@testing-library/react';
import MainHeader from '@/components/common/MainHeader';

jest.mock('next/navigation', () => ({
    useSelectedLayoutSegment: jest.fn(), // return undefined for base route
}))
 
it('renders mainHeader as Home when segment is Home', () => {
    render(<MainHeader />);

    const heading = screen.getByRole('heading', {
        name: /Home/i
    });
 
    expect(heading).toBeInTheDocument();
});

it('renders mainHeader as Friends when segment is Friends', () => {
    useSelectedLayoutSegment.mockReturnValueOnce('friends')
    render(<MainHeader />);
    const heading = screen.getByRole('heading', {
        name: /Friends/i
    });
 
    expect(heading).toBeInTheDocument();
});

it('renders correctly', () => {
    const { container } = render(<MainHeader />);
    expect(container).toMatchSnapshot();
});