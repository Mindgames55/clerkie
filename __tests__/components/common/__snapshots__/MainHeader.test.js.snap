// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <h1
      class="heading"
    >
      Home
    </h1>
  </div>
</div>
`;
